src/backend/ref/cipher-aesgcm.c

```diff
@@ -201,3 +201,82 @@ NoiseCipherState *noise_aesgcm_new_ref(void)
     state->parent.decrypt = noise_aesgcm_decrypt;
     return &(state->parent);
 }
+
+
+// Define getters on internal state
+// typedef struct
+// {
+//     struct NoiseCipherState_s parent;
+//     uint32_t aes[4 * (MAXNR + 1)];
+//     ghash_state ghash;
+//     uint8_t counter[16];
+//     uint8_t hash[16];
+
+// } NoiseAESGCMState;
+
+//#if defined(VAULT)
+
+
+uint8_t *get_aesgcm_counter(NoiseCipherState *state)
+{
+    NoiseAESGCMState *st = (NoiseAESGCMState *)state;
+    return st->counter;
+}
+
+
+uint8_t *get_aesgcm_hash(NoiseCipherState *state)
+{
+    NoiseAESGCMState *st = (NoiseAESGCMState *)state;
+    return st->hash;
+}
+
+
+uint8_t *get_aesgcm_aes(NoiseCipherState *state)
+{
+    NoiseAESGCMState *st = (NoiseAESGCMState *)state;
+    return (uint8_t *)(st->aes);
+}
+
+
+uint8_t *get_aesgcm_ghash_H(NoiseCipherState *state)
+{
+    NoiseAESGCMState *st = (NoiseAESGCMState *)state;
+    return (uint8_t*)(st->ghash.H);
+}
+
+
+uint8_t *get_aesgcm_ghash_Y(NoiseCipherState *state)
+{
+    NoiseAESGCMState *st = (NoiseAESGCMState *)state;
+    return (uint8_t*)(st->ghash.Y);
+}
+
+
+uint8_t get_aesgcm_ghash_posn(NoiseCipherState *state)
+{
+    NoiseAESGCMState *st = (NoiseAESGCMState *)state;
+    return st->ghash.posn;
+}
+
+size_t get_gcm_state_size(NoiseCipherState *state)
+{
+    NoiseAESGCMState *st = (NoiseAESGCMState *)state;
+    return sizeof(*st);
+}
+
+
+void set_aes_gcm_functions(NoiseCipherState *state)
+{
+    NoiseAESGCMState *st = (NoiseAESGCMState *)state;
+    // state->parent.cipher_id = NOISE_CIPHER_AESGCM;
```

Comment:
Are those commented lines mandatory? Moreover, why are they commented?

Reply:
I don't know, looks like you added this function according to commit 4dd46ae
But maybe this is copy pasted from my code?
I would say that this is important to be set as if I remember correctly, it is used outside the backend.
I can't explain why it works without this